example-usage/README.md

```diff
@@ -0,0 +1,92 @@
+   - **Linux:** Use your distro's package manager. E.g. on Ubuntu/Debian: `sudo apt-get update && sudo apt-get install python3-minimal gcc g++ make`
+   - **macOS:** Install the XCode Command Line Tools ([more info](https://github.com/nodejs/node-gyp/blob/main/README.md#on-macos))
+3. Make sure you have an OpenSSH compliant `ssh` command available and in your path if you plan to use the `Vim via SSH` example (it should already be there on macOS, and in Linux/WSL, you can install `openssh-client` using your distro's package manager if its missing)
+3. Install the latest dev container CLI: `npm install -g @devcontainers/cli`
+
+## Using the examples
+
+All examples use the contents of the `workspace` folder for their configuration, which is where you can modifications if you'd like. The example scripts are then in different sub-folders. 
+
+### Tool examples
+
+You can use these examples by opening a terminal and typing one of the following:
+
+- `tool-vscode-server/start.sh` - [VS Code Server](https://code.visualstudio.com/docs/remote/vscode-server) (official)
+- `tool-openvscode-server/start.sh` - [openvscode-server](https://github.com/gitpod-io/openvscode-server)
+- `tool-vim-via-ssh/start.sh` - Vim via an SSH connection. SSH is used primarily to demonstrate how this could be achieved from other SSH supporting client tools.
+
+When switching between examples, pass `true` in as an argument to get the container recreated to avoid port conflicts. e.g., `./start.sh true`
+
+In the first two examples, you'll be instructed to go to `http://localhost:8000` in a browser.
+
+This also adds a desktop to the container that can be accessed from a web browser at `http://localhost:6080` and you can connect using the password `vscode`.
+
+#### How the tool examples work
+
+These examples demonstrate the use of the dev container CLI to:
+
+1. Simplify setup using the "[dev container features](https://containers.dev/implementors/features/#devcontainer-json-properties)" concept. For example, SSH support is added just using a feature reference. See `workspace/.devcontainer/devcontainer.json` for more information.
+
+2. How the dev container CLI can be used to inject tools without building them into the base image:
+
+    1. Use `devcontainer up` to spin up the container and mount a `server` and `workspace` folder into the container.
+    2. Use `devcontainer exec` to run a script from this mounted folder to set up the appropriate server.
+    3. In the `vim` example, a temporary SSH key is set up and configured, and then SSH is used from the command line to connect to the container once it is up and running. See `tool-vim-via-ssh/start.sh` for details.
+
+Currently the `appPort` property is used in `devcontainer.json` instead of `forwardPorts` due to a gap in the current dev container CLI ([see here](https://github.com/devcontainers/cli/issues/22)).
+
+### CI build environment example
+
+You can use the example by opening a terminal and typing the following:
```

Comment:
Add a link to the CI action, so GitHub users are guided towards that.

Reply:
Updated!